### PR TITLE
feat(filelog): File Sorting and Reading of Only Current File 

### PR DESCRIPTION
**Pull request by Mrod1598**
### Proposed Change

- Add New UI Component for FileLogFiltering 
- Update parameter handling to validate filelog 
- Update filelog processor
- Update Resolver to Add Support for the new param

image

note:
~~There is currently a bug with timestamp parsing so I've commented out that section until the fix gets in.~~

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: model/parameter.go [hunk 1]
```diff
@@ -706,6 +709,50 @@ func (p ParameterDefinition) validateAwsCloudwatchNamedFieldType(_ parameterFiel
 	return nil
 }
 
+func (p ParameterDefinition) validateFileLogSortType(_ parameterFieldType, value any) error {
+	reflectValue := reflect.ValueOf(value)
+	kind := reflectValue.Kind()
+	if kind != reflect.Slice {
```
Comment (cpheps) [inline model/parameter.go:715]: What is the expected value of kind? I'm curious if we can achieve this with a type switch rather than reflection.

Comment (Mrod1598) [reply model/parameter.go]: Yeah I was trying to be consistent with what we are doing for AWS but you're probably right that removing reflection where we can is better.

**Comment by dsvanlani** [comment]
I was also able to save an invalid config by setting the sort type to alphabetical and I was not required to specify a Regex Key value:

image

File: ui/src/components/ResourceConfigForm/ParameterInput/ParameterInput.tsx [hunk 2]
```diff
@@ -148,8 +151,26 @@ export const ParameterInput: React.FC<{
...
+      case ParameterType.FileLogSort:
+        return (
+          <>
+            <FileLogSortInput
+              definition={definition}
+              value={formValues[definition.name]}
+              onValueChange={onValueChange}
+              readOnly={readOnly}
+            />
+          </>
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply ui/src/components/ResourceConfigForm/ParameterInput/ParameterInput.tsx]: Updated

File: ui/src/components/ResourceConfigForm/ParameterInput/ParameterInput.tsx [hunk 3]
```diff
@@ -26,7 +27,9 @@ export interface ParamInputProps<T> {
 export const ParameterInput: React.FC<{
   definition: ParameterDefinition;
   readOnly?: boolean;
-}> = ({ definition, readOnly }) => {
+  onValueChangeArg?: (v: any) => void;
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply ui/src/components/ResourceConfigForm/ParameterInput/ParameterInput.tsx]: Updated

**Review by dsvanlani — APPROVED** [review]

File: model/parameter.go [hunk 4]
```diff
@@ -706,6 +709,72 @@ func (p ParameterDefinition) validateAwsCloudwatchNamedFieldType(_ parameterFiel
 	return nil
 }
 
+func (p ParameterDefinition) validateFileLogSortType(_ parameterFieldType, value any) error {
+
```
[the remark this answers is not in the document]
Comment (cpheps) [reply model/parameter.go:713]: Don't submit a change just for this unless you have others to group in.

File: ui/src/components/ResourceConfigForm/ParameterInput/FileLogSortInput.tsx [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+          {
+            name: "location" + index,
+            label: "Location",
+            type: ParameterType.String,
+            required: false,
+            description: "The sort timezone location.",
+            options: {
+              gridColumns: 12,
+            },
+          },
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply ui/src/components/ResourceConfigForm/ParameterInput/FileLogSortInput.tsx]: Updated

**Review by cpheps — APPROVED** [review]

**Review by dsvanlani — CHANGES_REQUESTED** [review]
After the recent changes I'm seeing console errors again.  We discussed a strategy to fix this by using the pattern in the AWS parameter input.

**Review by dsvanlani — APPROVED** [review]

**Pull request merged**